src/SAML2/Assertion/Transformer/DecodeBase64Transformer.php

```diff
@@ -87,6 +77,23 @@ public function transform(Assertion $assertion): Assertion
     }
 
 
+    /**
+     * @param \SimpleSAML\SAML2\XML\saml\AttributeValues[] $encodedValues
+     * @return array
+     */
+    private function getDecodedAttributeValues(array $encodedValues): array
+    {
+        array_walk($encodedValues, function(&$v, $k) {
+            if (is_string($v)) {
+                foreach ($this->decodeValue($v) as $decoded) {
+                    $v[$k] = new AttributeValue($decoded);
+                }
+            }
+        });
+        return $encodedValues;
+    }
+
+
```

Comment:
@MKodde I'm a bit rusty and I can't seem to get this 'fix' to work..  What am I missing here?

Reply:
Sorry, Jaime and I spent all night on this and we were left frustrated.. I rolled back eventually